### PR TITLE
Feature/toolbar compose

### PR DESCRIPTION
**Pull request by MaxAstin**

### CODE REVIEW

File: app/src/main/java/com/bunbeauty/papakarlo/common/ui/toolbar/FoodDeliveryToolbarActions.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.bunbeauty.papakarlo.common.ui.toolbar
+
+import androidx.annotation.DrawableRes
+import com.bunbeauty.papakarlo.feature.top_cart.TopCartUi
+
+sealed class FoodDeliveryToolbarActions
```
Comment (xidealo) [inline app/src/main/java/com/bunbeauty/papakarlo/common/ui/toolbar/FoodDeliveryToolbarActions.kt:6]: interface?

File: app/src/main/java/com/bunbeauty/papakarlo/di/UIMapperModule.kt [hunk 2]
```diff
@@ -21,4 +22,9 @@ fun uiMapperModule() = module {
             stringUtil = get()
         )
     }
+    single {
```
Comment (xidealo) [inline app/src/main/java/com/bunbeauty/papakarlo/di/UIMapperModule.kt:25]: factory

File: app/src/main/java/com/bunbeauty/papakarlo/feature/product_details/ProductDetailsFragment.kt [hunk 3]
```diff
@@ -2,182 +2,253 @@ package com.bunbeauty.papakarlo.feature.product_details
...
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        viewBinding.fragmentProductDetailsCvMain.compose {
-            val menuProduct by viewModel.menuProduct.collectAsState()
-            ProductDetailsScreen(menuProduct)
+        viewModel.getMenuProduct(args.menuProductUuid)
+
+        viewBinding.fragmentProductDetailsCvMain.setContentWithTheme {
+            val menuProductUiState by viewModel.menuProductUiState.collectAsState()
```
Comment (xidealo) [inline app/src/main/java/com/bunbeauty/papakarlo/feature/product_details/ProductDetailsFragment.kt:65]: collectAsStateWithLifecycle

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/product_details/ProductDetailsViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+    private val mutableProductDetailsState = MutableStateFlow(ProductDetailsState())
+    val menuProductUiState = mutableProductDetailsState.asCommonStateFlow()
+
+    init {
+        observeCart()
+    }
+
+    fun getMenuProduct(menuProductUuid: String) {
+        sharedScope.launch {
```
Comment (xidealo) [inline shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/product_details/ProductDetailsViewModel.kt:29]: mb exceptionhandler?

File: app/src/main/java/com/bunbeauty/papakarlo/common/ui/toolbar/FoodDeliveryToolbarActions.kt (hunk 1 above)
Comment (MaxAstin) [reply app/src/main/java/com/bunbeauty/papakarlo/common/ui/toolbar/FoodDeliveryToolbarActions.kt]: fixed

File: app/src/main/java/com/bunbeauty/papakarlo/feature/product_details/ProductDetailsFragment.kt (hunk 3 above)
Comment (MaxAstin) [reply app/src/main/java/com/bunbeauty/papakarlo/feature/product_details/ProductDetailsFragment.kt]: fixed

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/product_details/ProductDetailsViewModel.kt (hunk 4 above)
Comment (MaxAstin) [reply shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/product_details/ProductDetailsViewModel.kt]: fixed

File: app/src/main/java/com/bunbeauty/papakarlo/di/UIMapperModule.kt (hunk 2 above)
Comment (MaxAstin) [reply app/src/main/java/com/bunbeauty/papakarlo/di/UIMapperModule.kt]: fixed

**Pull request merged**